### PR TITLE
Fix accessibility issues

### PR DESCRIPTION
**Pull request by alexandrevryghem**
## References
* Resolves part of #1191 (470941)
* Resolves part of #1174 (468404, 468405)
* Resolves part of #1186 (470748)
* Resolves part of #1178 (469493, 469644)
* Resolves part of #1175 (468900, 468945, 469023, 469026) (one the last 2 issues was already fixed)
* Fixes most of the remaining issue from #2055

## Description
This PR fixes many accessibility bugs, such as missing ARIA tags, absent roles, and incorrect header ordering. I have seperated the fixes by page/type of issue to facilitate the review process.

## Instructions for Reviewers
I used the Axe DevTools plugin and the Lighthouse plugin on most of the pages to find and address numerous issues. I did not fix any of the color contrast issues, as these are already being resolved by #2587, except for the ones on the processes, which were not covered in that particular PR.

## Checklist
- [x] My PR is small in size (e.g. less than 1,000 lines of code, not including comments & specs/tests), or I have provided reasons as to why that's not possible.
- [x] My PR passes [ESLint](https://eslint.org/) validation using `yarn lint`
- [x] My PR doesn't introduce circular dependencies (verified via `yarn check-circ-deps`)
- [x] My PR includes [TypeDoc](https://typedoc.org/) comments for _all new (or modified) public methods and classes_. It also includes TypeDoc for large or complex private methods.
- [x] My PR passes all specs/tests and includes new/updated specs or tests based on the [Code Testing Guide](https://wiki.lyrasis.org/display/DSPACE/Code+Testing+Guide).
- [x] If my PR includes new libraries/dependencies (in `package.json`), I've made sure their licenses align with the [DSpace BSD License](https://github.com/DSpace/DSpace/blob/main/LICENSE) based on the [Licensing of Contributions](https://wiki.lyrasis.org/display/DSPACE/Code+Contribution+Guidelines#CodeContributionGuidelines-LicensingofContributions) documentation.
- [x] If my PR includes new features or configurations, I've provided basic technical documentation in the PR itself.
- [x] If my PR fixes an issue ticket, I've [linked them together](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

### CODE REVIEW

File: src/app/footer/footer.component.html [hunk 1]
```diff
@@ -62,10 +62,11 @@ <h5 class="text-uppercase">Footer Content</h5>
         <a class="text-white"
            href="https://www.lyrasis.org/">{{ 'footer.link.lyrasis' | translate}}</a>
       </p>
-      <ul class="footer-info list-unstyled small d-flex justify-content-center mb-0">
+      <ul class="footer-info list-unstyled d-flex justify-content-center mb-0">
         <li>
-            <a class="text-white" href="javascript:void(0);"
-             (click)="showCookieSettings()">{{ 'footer.link.cookies' | translate}}</a>
+          <button class="btn btn-link text-white" type="button" (click)="showCookieSettings()">
```
Comment (tdonohue) [inline src/app/footer/footer.component.html:67]: @alexandrevryghem : I'm still reviewing/testing this PR.  So far it looks great!  However, I have a question on commit https://github.com/DSpace/dspace-angular/pull/2683/commits/9ab30231aad9983cc2d961e639e4a12a22fe1633

That commit almost seems to go *against* normal accessibility (an example is on this line). Accessibility tends to say you should use the native elements as much as possible... and in this situation the native element is a link (`<a>`) but you've made a `<button>` that looks like a link.

If we want to go this route, these buttons all MUST have `role="link"` per https://accessibleweb.com/question-answer/is-it-ok-to-have-a-link-that-looks-like-a-button/

But overall, I'm not sure these should be made into buttons. See the accessibility descriptions between buttons and links: https://www.w3schools.com/accessibility/accessibility_buttons_links.php

Is there a reason why you chose to make "uncrawlable links" into buttons?  Maybe I'm missing how this somehow helps accessibility?

Comment (tdonohue) [reply src/app/footer/footer.component.html:67]: Although... hmm.. right after I posted that question, I took a second look.  I think I see what you are doing.  You are only turning `<a>` into `<button>` when the link _is really an action that keeps you on the same page_.  So, that would be in line with the W3Schools definition above.  I may have convinced myself this is correct, but I'll leave this comment here in case others disagree.

Comment (mwoodiupui) [reply src/app/footer/footer.component.html:67]: I'd say that both are confusing but the replacement is moving in the right direction.  The old code is a link which looks like a link but is hijacked to act like a button.  The new code is a button which acts like a button but looks like a link.  The old code is confusing to screen readers, while the new code is visually confusing.  If it acts as a button, it should be coded `<button/>` and look like a button.

Comment (tdonohue) [reply src/app/footer/footer.component.html:67]: @mwoodiupui : While I understand your point, I think the UI will look awkward if we make these all look like buttons.  In this specific scenario (where these inline comments are), it's modifying the links in our default footer (see footer at https://demo.dspace.org).

By default there are 4 links there:
* Cookie settings (changed to a `<button>` in this PR, as you don't leave the page you are on to edit cookie settings)
* Privacy Policy (remains an `<a>` link, as it sends you to a different page)
* End User Agreement (remains an `<a>` link, as it sends you to a different page)
* Send Feedback (remains an `<a>` link, as it sends you to a different page)

Changing "Cookie Settings" to visually look like a button is awkward in this footer location.  It would be possible, but would likely require further redesign of the footer (which is not the intent of this PR).

Overall, in summary, I now see why @alexandrevryghem made the changes in https://github.com/DSpace/dspace-angular/commit/9ab30231aad9983cc2d961e639e4a12a22fe1633.  I'm OK with them. I do agree there may be more we can do here, but it likely is out-of-scope for this PR.

**Review by tdonohue — APPROVED** [review]
👍 Thank you so much @alexandrevryghem !  I went through all 19 commits one by one and verified that each page's accessibility warnings (from the Deque Axe DevTools plugin) are *fixed* as described.  Also checked the code and everything looks good as well. 

I've also double checked the referenced tickets, and the issues you referenced all appear to be fixed.  In some situations, that essentially "closes" the old ticket.  In others, there's still one or two issues on those pages, but I'll make that clearer in the tickets themselves.

Overall, this is a significant amount of accessibility fixes, so it's much appreciated! I'll make sure it is automatically backported to 7.x (via the `port to dspace-7_x` label), so that these fixes can all be included in 7.6.2.

**Comment by tdonohue** [comment]
@alexandrevryghem : After finishing my review, I was double checking the referenced tickets.  I wanted to note that I **don't believe this PR fixes 470941 in #1191:
> "Non-decorative content is inserted using CSS pseudo-elements.". Every button on this page (Upload, download, edit, delete, undo) is represented by an icon, but those icons are inserted using CSS :before and :after pseudo-element.

I believe the bug in that case is that these buttons **disappear completely** if you turn off CSS.  I can still reproduce that using the WAVE plugin for Chrome (if I turn off styles on that Item Edit -> Bitstreams tab.  The action buttons on bitstreams still appear empty & it's difficult to understand them.

I think the problem is likely that we aren't adding extra accessibility for these FontAwesome icons like is described here: https://fontawesome.com/docs/web/dig-deeper/accessibility#icons-used-as-semantic-elements-2

In other words, I think we'd need to have an invisible `<span>` to describe each button (for when CSS is turned off).   We can move this to a different PR though, if you'd rather. I just wanted to point out that I think this bug still needs resolving.

**Comment by alexandrevryghem** [comment]
@tdonohue: I fixed the issue. At first, I addressed it using the `fa-sr-only` class that you mentioned. However, Wave flagged additional warnings because the display text and the title attribute had the same value. Therefore, I opted to use the `aria-label` attributes instead (which is also more commonly used in the codebase). This fix also solves the problem when you disable CSS, so I believe this might be a better solution.

**Comment by tdonohue** [comment]
Thanks @alexandrevryghem !  I've verified today that the latest commits **do fix** 470941 in #1191.  I'll get this merged & ported to 7.x today (and update all the tickets where portions were fixed).

Thanks again so much for all the work to clean up accessibility.  This PR cleans up a ton of small issues to get us to *zero* (non-contrast issue) errors (in Axe and WAVE) on many pages. So, this combined with #2587 is a wonderful improvement!
